code/modules/food_and_drink/cakes.dm

```diff
@@ -24,62 +24,482 @@
+		c.set_loc(user)
+		src.clayer++
+		src.reagents.maximum_volume += 100
+		src.amount += 10
+		c.reagents.trans_to(src,c.reagents.total_volume)
+
+		for(var/i=1,i<=c.food_effects.len,i++) //adding food effects to the src that arent already present
+			if(c.food_effects[i] in src.food_effects)
+				continue
+			src.food_effects += c.food_effects[i]
+
+		var/singlecake //logging the clayer before changing it later
+		if(c.clayer == 1)
+			singlecake = 1
+
+		for(var/i=1,i<=src.overlays.len,i++) //looking for candles and if you set a cake on top of it, it pops off!
+			if(("[src.overlay_refs[i]]" == "cake[(src.clayer)-1]-candle") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-1]-candle_lit") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-2]-candle") || ("[src.overlay_refs[i]]" == "cake[(src.clayer)-2]-candle_lit"))
+				src.ClearSpecificOverlays("[src.overlay_refs[i]]")
+				var/obj/item/device/light/candle/can = new /obj/item/device/light/candle/small
+				can.set_loc(get_turf(src.loc))
+				user.show_text("<b>The candle pops off! Oh no!</b>","red")
+				cake_candle = 0
+				if(src.litfam)
+					src.put_out()
+				break
+
+		var/staticiterator = c.overlays.len
+		for(var/i=1,i<=staticiterator,i++) //the handling for actually adding the toppings to the cake
+			if(("[c.overlay_refs[i]]" == "base") || ("[c.overlay_refs[i]]" == "second")) //setting up base layer overlay
+				var/overlay_layer
+				if("[c.overlay_refs[i]]" == "base")
+					if(src.clayer == 2)
+						overlay_layer = "second"
+						src.amount2 = c.amount
+					else if(src.clayer == 3)
+						overlay_layer = "third"
+						src.amount3 = c.amount
+				else if("[c.overlay_refs[i]]" == "second")
+					overlay_layer = "third"
+					src.amount3 = c.amount
+					src.clayer++
+					src.reagents.maximum_volume += 100
+					src.amount += 10
+				var/image/stack = new /image('icons/obj/foodNdrink/food_dessert.dmi',"cake[src.clayer]-overlay")
+				stack.color = (c.GetOverlayImage(c.overlay_refs[i])).color
+				src.UpdateOverlays(stack,"[overlay_layer]")
+				continue
+			var/image/buffer = c.GetOverlayImage("[c.overlay_refs[i]]") //generating the topping reference from the original cake to be stacked
+			var/list/tag
+			if(src.clayer == 2)
+				tag = replacetext("[c.overlay_refs[i]]","1","2")
+			else if(src.clayer == 3 && singlecake)
+				tag = replacetext("[c.overlay_refs[i]]","1","3")
+			else
+				tag = replacetext("[c.overlay_refs[i]]","2","3")
+			var/image/newoverlay = new /image('icons/obj/foodNdrink/food_dessert.dmi',"[tag]")
+			if(buffer.color)
+				newoverlay.color = buffer.color
+			src.UpdateOverlays(newoverlay,"[tag]")
+		if(c.litfam)
+			src.ignite()
+		qdel(c)
+
+
+	proc/build_cake(var/obj/item/cake_transfer,var/mob/user,var/mode,var/layer_tag,var/replacetext)//cake_transfer : passes a reference to the cake that we are building //mode : 1 or 2 : cake or slice //layer_tag and replacetext : references used with slicing //decompiles a full cake into slices or other cakes
```

Comment:
It'd probably be better if you had something like
```
#define CAKE_MODE_CAKE 1
#define CAKE_MODE_SLICE 2
```
somewhere at the beginning of this file to avoid mysterious magic constants `1` and `2` in the code.